library/std/src/net/ip.rs

```diff
@@ -958,18 +948,21 @@ impl PartialOrd<IpAddr> for Ipv4Addr {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl Ord for Ipv4Addr {
     fn cmp(&self, other: &Ipv4Addr) -> Ordering {
-        u32::from_be(self.inner.s_addr).cmp(&u32::from_be(other.inner.s_addr))
+        // Compare as native endian `u32`: LE u32 in LE target, BE u32 in BE target.
+        // In LE target, `u32::from_be` swaps bytes of already-BE inner to make it a LE `u32`.
+        // In BE target, `u32::from_be` does nothing so we have BE `u32`.
```

Comment:
```suggestion
        // Compare as native endian `u32`: LE u32 in LE target, BE u32 in BE target.
```
I think the function names and saying "compare as native endian" are already clear enough. I find the next two lines confusing as they are about the implementation details of how you convert between endianesses, which is not relevant for understanding this code.